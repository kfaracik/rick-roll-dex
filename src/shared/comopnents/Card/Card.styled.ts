import {StyleSheet} from 'react-native';
import {Colors} from '../../utils';

export const styles = StyleSheet.create({
  card: {
    minHeight: 224,
    borderRadius: 24,
    borderColor: Colors.primary,
    borderWidth: 1,
    overflow: 'hidden',
    marginHorizontal: 16,
    padding: 12,
    margin: 0,
    boxShadow: [
      {
        offsetX: 5,
        offsetY: 5,
        color: Colors.primary,
      },
    ],
  },
  content: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    gap: 20,
    paddingHorizontal: 0,
    paddingVertical: 0,
    margin: 0,
  },
});
