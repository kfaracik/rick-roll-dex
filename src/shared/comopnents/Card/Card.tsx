import React, {ReactNode} from 'react';
import {ViewStyle} from 'react-native';
import {Card as CardRNP} from 'react-native-paper';
import {styles} from './Card.styled';

type CardProps = {
  onPress?: () => void;
  style?: ViewStyle;
  children: ReactNode;
};

export const Card = ({onPress, style, children}: CardProps) => {
  return (
    <CardRNP onPress={onPress} style={[styles.card, style]}>
      <CardRNP.Content style={styles.content}>{children}</CardRNP.Content>
    </CardRNP>
  );
};
