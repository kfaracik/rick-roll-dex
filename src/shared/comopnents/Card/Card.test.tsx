import React from 'react';
import {render} from '@testing-library/react-native';
import {Card} from './Card';
import {Text} from 'react-native-paper';

describe('Card component', () => {
  it('should generate snapshot', () => {
    const {toJSON} = render(
      <Card>
        <Text>Test texts</Text>
      </Card>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
