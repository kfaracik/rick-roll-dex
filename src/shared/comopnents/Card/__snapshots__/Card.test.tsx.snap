// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should generate snapshot 1`] = `
<View
  collapsable={false}
  entering={
    FadeIn {
      "build": [Function],
      "durationV": 300,
      "randomizeDelay": false,
      "reduceMotionV": "system",
    }
  }
  exiting={
    FadeOut {
      "build": [Function],
      "durationV": 300,
      "randomizeDelay": false,
      "reduceMotionV": "system",
    }
  }
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 24,
        "margin": 0,
        "marginHorizontal": 16,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
      }
    }
    testID="card-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(247, 243, 249, 1)",
          "borderColor": "#224229",
          "borderRadius": 24,
          "borderWidth": 1,
          "boxShadow": [
            {
              "color": "#224229",
              "offsetX": 5,
              "offsetY": 5,
            },
          ],
          "elevation": 1,
          "flex": undefined,
          "minHeight": 224,
          "overflow": "hidden",
          "padding": 12,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
      testID="card-container"
    >
      <View
        style={
          [
            {
              "flexShrink": 1,
            },
            undefined,
          ]
        }
        testID="card"
      >
        <View
          borderRadiusStyles={
            {
              "borderRadius": 24,
            }
          }
          style={
            [
              {
                "paddingHorizontal": 16,
              },
              {
                "paddingVertical": 16,
              },
              [
                {
                  "alignItems": "flex-start",
                  "gap": 20,
                  "margin": 0,
                  "paddingHorizontal": 0,
                  "paddingVertical": 0,
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Test texts
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Card component should generate snapshot 1`] = `
<View
  collapsable={false}
  entering={
    FadeIn {
      "build": [Function],
      "durationV": 300,
      "randomizeDelay": false,
      "reduceMotionV": "system",
    }
  }
  exiting={
    FadeOut {
      "build": [Function],
      "durationV": 300,
      "randomizeDelay": false,
      "reduceMotionV": "system",
    }
  }
  jestAnimatedStyle={
    {
      "value": {},
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 24,
        "margin": 0,
        "marginHorizontal": 16,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
      }
    }
    testID="card-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(247, 243, 249, 1)",
          "borderColor": "#224229",
          "borderRadius": 24,
          "borderWidth": 1,
          "boxShadow": [
            {
              "color": "#224229",
              "offsetX": 5,
              "offsetY": 5,
            },
          ],
          "elevation": 1,
          "flex": undefined,
          "minHeight": 224,
          "overflow": "hidden",
          "padding": 12,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
      testID="card-container"
    >
      <View
        style={
          [
            {
              "flexShrink": 1,
            },
            undefined,
          ]
        }
        testID="card"
      >
        <View
          borderRadiusStyles={
            {
              "borderRadius": 24,
            }
          }
          style={
            [
              {
                "paddingHorizontal": 16,
              },
              {
                "paddingVertical": 16,
              },
              [
                {
                  "alignItems": "flex-start",
                  "gap": 20,
                  "margin": 0,
                  "paddingHorizontal": 0,
                  "paddingVertical": 0,
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Test texts
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
